howso/utilities/feature_attributes/base.py

```diff
@@ -74,16 +79,91 @@ def get_parameters(self) -> dict:
         """
         return self.params
 
-    def to_json(self) -> str:
+    def to_json(self, archive: bool = False, json_path: t.Optional[Path] = None) -> str:
         """
         Get a JSON string representation of this FeatureAttributes object.
 
+        Parameters
+        ----------
+        archive : bool, default False
+            If True, the returned JSON includes 3 top-level keys:
+
+            - feature_attributes - A nested map of the inferred feature attributes.
+            - params - A map of parameters and their values used to infer the feature attributes.
+            - unsupported - A list of features not supported by the Howso Engine.
+
+            If False, only the nested map of the feature attributes is returned.
+
+        json_path : Path, optional
+            If provided, the JSON will be written to this path in addition to being returned.
+
         Returns
         -------
         String
             A JSON representation of the inferred feature attributes.
         """
-        return json.dumps(self)
+        if archive:
+            json_str = json.dumps({
+                "feature_attributes": self,
+                "params": self.params,
+                "unsupported": self.unsupported,
+            })
+        else:
+            json_str = json.dumps(self)
+
+        if json_path:
+            with open(json_path, mode="w") as fp:
+                fp.write(json_str)
+
+        return json_str
+
+    @classmethod
+    def from_json(
+        cls: type[FeatureAttributesBaseType],
+        json_str: t.Optional[str] = None,
+        *,
+        json_path: t.Optional[Path] = None
+    ) -> FeatureAttributesBaseType:
```

Comment:
Why does this need to be a separate TypeVar instead of just `FeatureAttributesBase` ? Not clear what is gained with this extra type since it is bound to FeatureAttributesBase.